packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.scss

```diff
@@ -15,32 +15,31 @@
   @include right(9px);
 }
 
-:global {
-  .ms-DatePicker {
-    @include ms-u-normalize;
-    margin-bottom: 17px;
-
-    .ms-TextField {
-      position: relative;
-
-      input::-ms-clear {
-        display: none;
-      }
-
-      input[readonly] {
-        cursor: pointer;
-      }
-    }
-  }
+.root {
+  @include ms-u-normalize;
+  margin-bottom: 17px;
+}
 
-  // Insert a calendar icon over the date field.
-  .ms-DatePicker-event--with-label {
-    @include ms-DatePicker-event();
-    top: 35px;
+.textField {
+  position: relative;
+
+  input::-ms-clear {
+    display: none;
   }
 
-  .ms-DatePicker-event--without-label {
-    @include ms-DatePicker-event();
-    top: 7px;
+  input[readonly] {
+    cursor: pointer;
   }
 }
+
+// Insert a calendar icon over the date field.
+.eventWithLabel {
```

Comment:
Maybe `.eventHasLabel` and `.eventHasNoLabel`. Doesn't sound as good but sticking to "is" or "has" for state/modifier classes will help them stand out.